ospf6d/ospf6_top.c

```diff
@@ -865,88 +866,203 @@ DEFUN (no_ospf6_stub_router_shutdown,
 }
 #endif
 
-static void ospf6_show(struct vty *vty, struct ospf6 *o)
+
+static void ospf6_show(struct vty *vty, struct ospf6 *o, json_object *json,
+		       bool use_json)
 {
 	struct listnode *n;
 	struct ospf6_area *oa;
 	char router_id[16], duration[32];
 	struct timeval now, running, result;
 	char buf[32], rbuf[32];
+	json_object *json_areas = NULL;
+
+	if (use_json) {
+		json_areas = json_object_new_object();
+
+		/* process id, router id */
+		inet_ntop(AF_INET, &o->router_id, router_id, sizeof(router_id));
+		json_object_string_add(json, "routerId", router_id);
+
+		/* running time */
+		monotime(&now);
+		timersub(&now, &o->starttime, &running);
+		timerstring(&running, duration, sizeof(duration));
+		json_object_string_add(json, "running", duration);
+
+		/* Redistribute configuration */
+		/* XXX */
+		json_object_int_add(json, "lsaMinimumArrivalMsecs",
+				    o->lsa_minarrival);
+
+		/* Show SPF parameters */
+		json_object_int_add(json, "spfScheduleDelayMsecs",
+				    o->spf_delay);
+		json_object_int_add(json, "holdTimeMinMsecs", o->spf_holdtime);
+		json_object_int_add(json, "holdTimeMaxMsecs",
+				    o->spf_max_holdtime);
+		json_object_int_add(json, "holdTimeMultplier",
+				    o->spf_hold_multiplier);
+
+
+		if (o->ts_spf.tv_sec || o->ts_spf.tv_usec) {
+			timersub(&now, &o->ts_spf, &result);
+			timerstring(&result, buf, sizeof(buf));
+			ospf6_spf_reason_string(o->last_spf_reason, rbuf,
+						sizeof(rbuf));
+			json_object_string_add(json, "spfLastExecutedMsecs",
+					       buf);
+			json_object_string_add(json, "spfLastExecutedReason",
+					       rbuf);
+
+			json_object_int_add(
+				json, "spfLastDurationSecs",
+				(long long)o->ts_spf_duration.tv_sec);
+
+			json_object_int_add(
+				json, "spfLastDurationMsecs",
+				(long long)o->ts_spf_duration.tv_usec);
+		} else
+			json_object_boolean_true_add(json, "spfHasNotRun");
```

Comment:
I don't like this pattern of conditionally including fields. If there's a field it should always be present, especially for a boolean field. I also think this should be `spfHasRun`.